Mage/src/main/java/mage/abilities/keyword/BlitzAbility.java

```diff
@@ -38,7 +38,7 @@ public BlitzAbility(Card card, String manaString) {
         );
         ability.addEffect(new GainAbilitySourceEffect(new DiesSourceTriggeredAbility(
                 new DrawCardSourceControllerEffect(1)
-        ).setTriggerPhrase("When this creature dies, ")));
+        ).setTriggerPhrase("When this creature dies, "), Duration.EndOfTurn));
```

Comment:
I think you can't just make it last until end of turn. If you have something that lets you skip the sacrifice, the creature still needs haste and the "when this creature dies" ability.